From c2e59a0a266e61e6688311df9aadadbb0eaf997b Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Thu, 26 Sep 2013 13:40:03 +0200
Subject: [PATCH 8/9] [Qt] RepaintRequested signal sometimes not emitted
 https://bugs.webkit.org/show_bug.cgi?id=121790

Reviewed by Simon Hausmann.

We should also emit repaintRequested after calling PageClient::update().

* WidgetSupport/PageClientQt.cpp:
(WebCore::PageClientQWidget::repaintViewport):
(WebCore::PageClientQGraphicsWidget::repaintViewport):

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@156329 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 Source/WebKit/qt/WebCoreSupport/PageClientQt.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Source/WebKit/qt/WebCoreSupport/PageClientQt.cpp b/Source/WebKit/qt/WebCoreSupport/PageClientQt.cpp
index 9979401..a33fb1c 100644
--- a/Source/WebKit/qt/WebCoreSupport/PageClientQt.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/PageClientQt.cpp
@@ -162,6 +162,7 @@ void PageClientQWidget::syncLayers(Timer<PageClientQWidget>*)
     if (TextureMapperLayerClient->rootLayer()->descendantsOrSelfHaveRunningAnimations() && !syncTimer.isActive())
         syncTimer.startOneShot(1.0 / 60.0);
     update(view->rect());
+    QMetaObject::invokeMethod(page, "repaintRequested", Qt::QueuedConnection, Q_ARG(QRect, view->rect()));
 }
 #endif
 
@@ -294,6 +295,7 @@ void PageClientQGraphicsWidget::syncLayers()
     if (TextureMapperLayerClient->rootLayer()->descendantsOrSelfHaveRunningAnimations() && !syncTimer.isActive())
         syncTimer.startOneShot(1.0 / 60.0);
     update(view->boundingRect().toAlignedRect());
+    QMetaObject::invokeMethod(page, "repaintRequested", Qt::QueuedConnection, Q_ARG(QRect, view->boundingRect().toAlignedRect()));
 }
 
 void PageClientQGraphicsWidget::setRootGraphicsLayer(GraphicsLayer* layer)
-- 
1.8.3.1

